File: capibara-website/app/models/Struct.scala
package models
import play.api.libs.json._

object Field {
  implicit val fieldFormatter = Json.format[Field];
}

object Struct {
  implicit val structFormatter = Json.format[Struct];
}

case class Struct(
    name: String,
    header: HeaderRef,
    summary: String,
    fields: Seq[Field],
    description: String,
    os_affinity: Seq[String]
)

case class Field(
    name: String,
    `type`: String,
    description: String,
)

File: capibara-website/src/models/_struct.rs
use super::header::HeaderSummary;
use rocket::serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, PartialEq)]
pub(crate) struct Struct {
    pub name: Box<String>,
    pub header: HeaderSummary,
    pub summary: Box<String>,
    pub fields: Vec<Field>,
    pub description: Box<String>,
    pub os_affinity: Vec<String>,
}

#[derive(Serialize, Deserialize, Clone, PartialEq)]
pub(crate) struct Field {
    pub name: Box<String>,
    #[serde(rename = "type")]
    pub _type: Box<String>,
    pub description: Box<String>,
}
